### PR TITLE
Feature/currency job

### PR DESCRIPTION
**Pull request by HappyCyborg**
endpoints to update currency info/rates

### CODE REVIEW

File: api/src/main/java/com/github/bablo_org/bablo_project/api/utils/Resource.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.Objects;
+
+public class Resource {
+
+    public String getResource(String resource) {
+        StringBuilder json = new StringBuilder();
+        try {
+            BufferedReader in = new BufferedReader(
```
Comment (AVPcm) [inline api/src/main/java/com/github/bablo_org/bablo_project/api/utils/Resource.java:14]: `try (A = initA(); B = initB() {
    // work with A and B
}
// here A and B will be automatically closed regardless of failure or success
`

File: api/src/main/java/com/github/bablo_org/bablo_project/api/model/CurrencyAPI/CurrencyInfo.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package com.github.bablo_org.bablo_project.api.model.CurrencyAPI;
```
Comment (AVPcm) [inline api/src/main/java/com/github/bablo_org/bablo_project/api/model/CurrencyAPI/CurrencyInfo.java:1]: https://docs.oracle.com/javase/tutorial/java/package/namingpkgs.html#:~:text=Naming%20Conventions,a%20programmer%20at%20example.com%20.

File: api/src/main/java/com/github/bablo_org/bablo_project/api/model/CurrencyAPI/CurrencyRates.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import lombok.Value;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
+import static java.util.Optional.ofNullable;
+
+@Value
+public class CurrencyRates {
```
Comment (AVPcm) [inline api/src/main/java/com/github/bablo_org/bablo_project/api/model/CurrencyAPI/CurrencyRates.java:12]: `CurrencyRate`

File: api/src/main/java/com/github/bablo_org/bablo_project/api/model/CurrencyAPI/CurrencyRatesToken.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package com.github.bablo_org.bablo_project.api.model.CurrencyAPI;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+public class CurrencyRatesToken {
```
Comment (AVPcm) [inline api/src/main/java/com/github/bablo_org/bablo_project/api/model/CurrencyAPI/CurrencyRatesToken.java:10]: `CurrencyRatesApiToken`

File: api/src/main/java/com/github/bablo_org/bablo_project/api/controller/CurrencyController.java [hunk 5]
```diff
@@ -18,9 +17,24 @@ public class CurrencyController extends BaseController {
...
     @GetMapping
     @ResponseBody
     List<Currency> getAll() {
         return service.getAll();
     }
+
+
+    @PutMapping("/updateRates")
+    void updateRates(){
+        service.updateRates(apiRates.getRates());
```
Comment (AVPcm) [inline api/src/main/java/com/github/bablo_org/bablo_project/api/controller/CurrencyController.java:33]: controllers shouldn't know about API clients, they contain only transport/routing logic

File: api/src/main/java/com/github/bablo_org/bablo_project/api/client/CurrencyInfoClient.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+import lombok.SneakyThrows;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@RequiredArgsConstructor
+@Component
+public class CurrencyInfoClient {
```
Comment (AVPcm) [inline api/src/main/java/com/github/bablo_org/bablo_project/api/client/CurrencyInfoClient.java:17]: we may create interfaces/impl for clients as well as for services

File: api/src/main/java/com/github/bablo_org/bablo_project/api/client/CurrencyRatesClient.java [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+        URL url = new URL(String.format(URL_STRING, token.getToken()));
+        HttpURLConnection request = (HttpURLConnection) url.openConnection();
+        request.connect();
+
+// Convert to JSON
+        JsonParser jp = new JsonParser();
+        InputStreamReader inputStreamReader = new InputStreamReader((InputStream) request.getContent());
+        JsonElement jsonRates = jp.parse(inputStreamReader);
+        request.disconnect();
+        inputStreamReader.close();
```
Comment (AVPcm) [inline api/src/main/java/com/github/bablo_org/bablo_project/api/client/CurrencyRatesClient.java:49]: if we will get an error between lines 46 and 49 then InputStream has a chance to not be closed, please use try-with-resources this way

File: api/src/main/java/com/github/bablo_org/bablo_project/api/utils/Resource.java (hunk 1 above)
Comment (HappyCyborg) [reply api/src/main/java/com/github/bablo_org/bablo_project/api/utils/Resource.java]: fixed

File: api/src/main/java/com/github/bablo_org/bablo_project/api/model/CurrencyAPI/CurrencyInfo.java (hunk 2 above)
Comment (HappyCyborg) [reply api/src/main/java/com/github/bablo_org/bablo_project/api/model/CurrencyAPI/CurrencyInfo.java]: fixed

File: api/src/main/java/com/github/bablo_org/bablo_project/api/utils/Resource.java (hunk 1 above)
Comment (HappyCyborg) [reply api/src/main/java/com/github/bablo_org/bablo_project/api/utils/Resource.java]: fixed

File: api/src/main/java/com/github/bablo_org/bablo_project/api/model/CurrencyAPI/CurrencyInfo.java (hunk 2 above)
Comment (HappyCyborg) [reply api/src/main/java/com/github/bablo_org/bablo_project/api/model/CurrencyAPI/CurrencyInfo.java]: fixed

File: api/src/main/java/com/github/bablo_org/bablo_project/api/model/CurrencyAPI/CurrencyRates.java (hunk 3 above)
Comment (HappyCyborg) [reply api/src/main/java/com/github/bablo_org/bablo_project/api/model/CurrencyAPI/CurrencyRates.java]: fixed

File: api/src/main/java/com/github/bablo_org/bablo_project/api/model/CurrencyAPI/CurrencyRatesToken.java (hunk 4 above)
Comment (HappyCyborg) [reply api/src/main/java/com/github/bablo_org/bablo_project/api/model/CurrencyAPI/CurrencyRatesToken.java]: fixed

File: api/src/main/java/com/github/bablo_org/bablo_project/api/controller/CurrencyController.java (hunk 5 above)
Comment (HappyCyborg) [reply api/src/main/java/com/github/bablo_org/bablo_project/api/controller/CurrencyController.java]: fixed

File: api/src/main/java/com/github/bablo_org/bablo_project/api/client/CurrencyInfoClient.java (hunk 6 above)
Comment (HappyCyborg) [reply api/src/main/java/com/github/bablo_org/bablo_project/api/client/CurrencyInfoClient.java]: fixed

File: api/src/main/java/com/github/bablo_org/bablo_project/api/client/CurrencyRatesClient.java (hunk 7 above)
Comment (HappyCyborg) [reply api/src/main/java/com/github/bablo_org/bablo_project/api/client/CurrencyRatesClient.java]: fixed

**Pull request closed without merging**